### ISSUE TITLE
blank webui on non default port

### ISSUE THREAD

**Issue by michel47** [ipfs/ipfs-webui#1758]
When I access the webui with the url : <http://127.0.0.1:8390/ipfs/Qmb3cY3zFJ5isjJ5H9cP47Vfqa6pqNwypbuo2TiBGjUmLd/> I get a blank page (my IPFS node is running on 8390/5311 ports)
same goes with the url : http://127.0.0.1:8390/ipns/webui.ipfs.io/

When I run it from http://127.0.0.1:5311/ipfs/bafybeif4zkmu7qdhkpf3pnhwxipylqleof7rl6ojbe7mq3fzogz6m4xk3i/ or <http://127.0.0.1:5001/ipfs/bafybeif4zkmu7qdhkpf3pnhwxipylqleof7rl6ojbe7mq3fzogz6m4xk3i/> and
<http://127.0.0.1:8080/ipfs/Qmb3cY3zFJ5isjJ5H9cP47Vfqa6pqNwypbuo2TiBGjUmLd/#/>
it runs fine,
however url that used to work don't anymore:
- <http://127.0.0.1:5001/ipfs/Qmb3cY3zFJ5isjJ5H9cP47Vfqa6pqNwypbuo2TiBGjUmLd/>
- <http://127.0.0.1:5001/ipns/webui.ipfs.io/>

The error message have is depicted below:
![screenshot](http://gateway.ipfs.io/ipfs/QmVPQrcxyL8GYYPpybHSfKbd72No3YAjvAi3cPpjVaSdqe)

**To Reproduce**
1.  setup your local ipfs gateway and api to the following : 
    -   Gateway: http://0.0.0.0:8390 and
    -   API : http://127.0.0.1:5311/
2. Go to <http://127.0.0.1:8390/ipfs/bafybeif4zkmu7qdhkpf3pnhwxipylqleof7rl6ojbe7mq3fzogz6m4xk3i/>
3. See errors on the console

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Browser firefox
 - Version 87
- my config file is as <http://127.0.0.1:8080/ipfs/Qma7FnT6NThQTxb4uZMGfhBCQjsicCsVDq25qaqAxfoKfc>

**Additional context**
it does work on Chromium and Brave

**Comment by lidel** [comment]
What you described is correct behavior.

Every go-ipfs release ships with a safe-listed version of webui, that was tested with specific version of the node.
Namely,  `[api-port]/webui` redirects to a safe-listed `[api-port]/ipfs/{cid}`  and that version is in the same Origin as API, so it does not require setting any CORS to work.

**TLDR**: you should use Webui via URL printed when your node starts, in your case I see custom API port `5311`, so the  URL you want to bookmark is  `http://127.0.0.1:5311/webui` (consider installing [companion extension](https://github.com/ipfs/ipfs-companion#readme), so you can open it via browser action).

If you want to use a different version that the one from the API port, that is on your own risk: you are on your own to set proper CORS headers etc. If you are ok with that, the latest release of this the webui can be found at https://webui.ipfs.io

**Comment by DrI-T** [comment]
@lidel thanks for your answer, I understand now : I need to use the API port for the webui
 still I am puzzled about why the URLs below don't work :

    * http://127.0.0.1:5001/ipfs/Qmb3cY3zFJ5isjJ5H9cP47Vfqa6pqNwypbuo2TiBGjUmLd/
    * http://127.0.0.1:5001/ipns/webui.ipfs.io/
    
    they both return ``404 page not found``

**Comment by lidel** [comment]
@DrI-T API port allows only "safelisted" CIDs to be loaded via `/ipfs/` and `/ipns/` is disabled.

FYI the list of safelisted CIDs is at https://github.com/ipfs/go-ipfs/blob/v0.9.0-rc1/core/corehttp/webui.go

Loading other versions of webui is possible via gateway port, but if you do that, need to safelist them using CORS mechanism.
